recipes/tests.rb

```diff
@@ -138,3 +138,68 @@
     JQMERGE
   end
 end
+
+# only test on GPU instances
+if node['cfncluster']['nvidia']['enabled'] == 'yes'
+  bash 'test nvidia driver install' do
+    cwd Chef::Config[:file_cache_path]
+    code <<-TESTNVIDIA
```

Comment:
I think it might be a good idea to have `set -e` in tests so that we fail if any command fails

Reply:
`set -e` on line 153 and line 182. If we do `set -e` before the line 147, in the case of the instance not having GPU hardware and we are not expected to run this test, the program will `exit 1` after line 147, which is something we don't want. Hence `set -e` only after the check for GPU hardware